server/etcdserver/api/membership/membership_test.go

```diff
@@ -39,34 +40,60 @@ func TestAddRemoveMember(t *testing.T) {
 	c.RemoveMember(17, true)
 	c.RemoveMember(18, true)
 
-	if false {
-		// TODO: Enable this code when Recover is reading membership from the backend.
-		c2 := newTestCluster(t, nil)
-		c2.SetBackend(be)
-		c2.Recover(func(*zap.Logger, *semver.Version) {})
-		assert.Equal(t, []*Member{{ID: types.ID(18),
-			Attributes: Attributes{Name: "node18"}}}, c2.Members())
-		assert.Equal(t, true, c2.IsIDRemoved(17))
-		assert.Equal(t, false, c2.IsIDRemoved(18))
-	}
+	c.AddMember(newTestMember(19, nil, "node19", nil), true)
```

Comment:
I guess we can get rid of the `shouldApplyV3` bool enum if membership is no longer stored in v2 store. https://github.com/etcd-io/etcd/pull/12854#issuecomment-819054199

Before that, how about changing all `true` to `ApplyBoth`?